Arrowgene.Ddon.GameServer/Handler/CraftRecipeGetCraftRecipeHandler.cs

```diff
@@ -23,15 +23,25 @@ public override void Handle(GameClient client, StructurePacket<C2SCraftRecipeGet
                 .Select(recipes => recipes.RecipeList)
                 .SingleOrDefault(new List<CDataMDataCraftRecipe>());
 
-            client.Send(new S2CCraftRecipeGetCraftRecipeRes()
+            foreach (CDataMDataCraftRecipe cDataMDataCraftRecipe in allRecipesInCategory)
+            {
+                // TODO: All furniture recipes by default should be hidden - check which furniture recipes have been unlocked for player via achievements
```

Comment:
This should probably be driven by the crafting recipe json but since its a todo it can stay like this

Reply:
The IsHide property is already part of the recipe JSON. By default we can set the property to true for furniture recipes in the JSON.
Then for each player we have to check which ones they have access to.

Reply:
Would it be a possibility for the server to control which recipe corresponds to which achievement or is that client driven?

Reply:
I think that's actually the expectation.
The client knows the achievement titles & info text. It also knows which achievement ID is associated with which icon, which obviously kind of sets the expectation to receive the item that is visible as per title/icon.

But the association of reward ID => recipe/item is performed by the server.

That also means we have to create a mapping of artificial reward IDs => item ID (this is the only thing sent by server in packets) / recipe ID